### PR TITLE
Feat/game policy domain extension

### PR DESCRIPTION
**Pull request by Bereket-Eshete**
Implement PRIORITIZED-OR Knob-Building for Game Strategies
## Description

The existing MOSES knob-building chain was designed for boolean logic (AND/OR/NOT) but needed adaptation for game strategies using PRIORITIZED-OR operators. Game moves don't have logical negation (no "NOT play-center") and use move reordering rather than boolean operations.

### Domain-Specific Function Design
Boolean logic and game strategies require fundamentally different approaches. For boolean logic, MOSES uses Logical Subtree Knobs (LSK) with multiplicity 3 to handle true/false/negation states. For game strategies, we needed a different knob type that reflects the nature of game moves - they can be included, excluded, or reordered, but not logically negated.

Where boolean logic uses `logicalProbe` to create LSK knobs with boolean operations (AND/OR/NOT), we created `strategyProbe` that generates Strategy Subtree Knobs (SSK) with multiplicity 2. The boolean system uses buildLogical for tree decoration, while game strategies use `buildStrategy` that understands `PRIORITIZED-OR` semantics. For permutation sampling, boolean logic relies on sampleLogicalPerms with logical transformations, whereas game strategies use sampleStrategiesPerms that focuses on move sequence variations.

This implementation extends MOSES's knob-building system to support both domains: boolean logic evolution continues to work with LSK knobs and logical operations, while game strategy evolution now works with SSK knobs and move reordering. The unified system allows MOSES to evolve both boolean expressions and tic-tac-toe move sequences through the same evolutionary framework with appropriate knob types and operations for each use case.

### CODE REVIEW

File: representation/sample-logical-perms.metta [hunk 1]
```diff
@@ -53,6 +53,49 @@
...
+        )
+        (if (<= $nPairs 1)
+            (let $move (car-atom $moveLabels) ($move $move)) ;; edge case: return duplicate for consistency
+            (let*
+                (
+                    ($upperbound (- (length $idxPairs) 1))
+                    ($picked (eval (lazyRandomSelector 0 $upperbound $nPairs)))
+                    ($perms (collapse (eval (getMoveArgs $op $idxPairs $picked $moveLabels))))
+                )
+                (eval (concatT $moveLabels $perms))
```
[the remark this answers is not in the document]
Comment (Bereket-Eshete) [reply representation/sample-logical-perms.metta]: @Yagth  I implemented this approach by mirroring the boolean logic pattern from sampleLogicalPerms. My approach was to adapt the proven sampling mechanism to the game strategy domain by replacing boolean operators with PRIORITIZED-OR and logical variables with game moves, while preserving the same structural flow.

File: representation/tests/sample-logical-perms-test.metta [hunk 2]
```diff
@@ -8,3 +8,21 @@
...
+
+;; Testcase for sample-strategy-perms
+
+;; Test with single move (edge case)
+!(assertEqual (length (sampleStrategyPerms PRIORITIZED-OR 1 (play-center))) 2)
+!(assertEqual (length (sampleStrategyPerms PRIORITIZED-OR 2 (play-center play-corner))) 4)
+!(assertEqual (length (sampleStrategyPerms PRIORITIZED-OR 3 (play-center play-corner play-block))) 6)
+;; Test with 4 moves
+!(assertEqual (length (sampleStrategyPerms PRIORITIZED-OR 4 (play-center play-corner play-block play-fork))) 8)
+
```
[the remark this answers is not in the document]
Comment (Yagth) [reply representation/tests/sample-logical-perms-test.metta]: On second view. The output length is two because we have something like `(PRIORITIZED-OR play-center)`. So that might not be a problem :)

File: representation/sample-logical-perms.metta (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bereket-Eshete) [reply representation/sample-logical-perms.metta]: @Yagth  ok i understand it now i will fix it. but i have one question  Should the sampling function generate individual moves instead of PRIORITIZED-OR expressions to avoid nesting? For example, instead of generating (PRIORITIZED-OR play-center play-corner), (PRIORITIZED-OR play-corner play-center)  should it generate just the sequence play-center play-corner.
